### ISSUE TITLE
Model Inference Instructions

### ISSUE THREAD

**Issue by tdsone** [fburic/protein-mgem#1]
Hey there,

what should I do if I just want to do use this model to predict protein abundance from AA sequence? 

Best
Timon

**Comment by fburic** [comment]
Hi Timon,

Sorry the repo isn't very user-friendly right now. The focus was to have the paper results reproducible, so scripts for common "end-user" routines aren't  present yet.

To predict abundance, you'll have to write a few lines of code. I will add a section to the main README on this, but for now you can take a look at https://github.com/fburic/protein-mgem/blob/main/results/20211223_182228/inspect_best_model.ipynb

The basic steps are below (slightly adapted lines from the notebook I mentioned).

**1. Load the model and get a sequence tokenizer**

```python
model = ProteinBertForValuePrediction.from_pretrained(bert_checkpoint_fname, output_attentions=True)
model.eval()
tokenizer = TAPETokenizer(vocab='iupac')
```

**2. Tokenize and evaluate amino acid sequences**

The model was trained on Box-Cox-transformed abundance values (molecules per cell), so remember to inverse-transform predictions if you need molecule counts. The Box-Cox lambda is below.

```python
predictions = []
for seq in tqdm(sequences):
    tokenized_seq = torch.tensor([tokenizer.encode(seq)])
    pred, _ = model(tokenized_seq)
    predictions.append(pred.detach().numpy()[0][0])
predictions = np.array(predictions)

BOXCOX_LAMBDA = -0.05155
predictions = special.inv_boxcox(predictions, BOXCOX_LAMBDA)
```
 (I know looping over single evaluations like this is terribly inefficient but padding wasn't implemented and for our evaluations this was fast enough).

For the imports, setting random seeds, and so on, please see the notebook.
You'll have to `pip install -e` the `protein_mgem` directory to have the codebase in your environment.

I'll try to make some time early next week to write a section for the README.
Hope this helps until then.

Best,
Filip

**Comment by fburic** [comment]
Hi again,

I have now added a section to the repo's README with a working code example for obtaining predictions.
This will become a script later on but leaving the code base as-is for now while the article is in review.

Best,
Filip

**Comment by tdsone** [comment]
Thanks a lot!!!
